instrumentation/rabbitmq-2.7/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/rabbitmq/RabbitMqTest.java

```diff
@@ -0,0 +1,955 @@
+                    }));
+  }
+
+  private static Stream<Arguments> provideParametersForMessageCountAndTimestamp() {
+    return Stream.of(
+        Arguments.of("some-exchange", 1, false),
+        Arguments.of("some-exchange", 2, false),
+        Arguments.of("some-exchange", 3, false),
+        Arguments.of("some-exchange", 4, false),
+        Arguments.of("some-exchange", 1, true),
+        Arguments.of("some-exchange", 2, true),
+        Arguments.of("some-exchange", 3, true),
+        Arguments.of("some-exchange", 4, true));
+  }
+
+  private static Stream<Arguments> provideParametersForCommandError() {
+    return Stream.of(
+        Arguments.of(
+            "exchange.declare",
+            "IOException",
+            null,
+            null,
+            (java.util.function.Consumer<Channel>)
+                channel -> {
+                  try {
+                    channel.exchangeDeclare("some-exchange", "invalid-type", true);
+                  } catch (Exception e) {
+                    throw new RuntimeException(e);
+                  }
+                }),
+        Arguments.of(
+            "Channel.basicConsume",
+            "IllegalStateException",
+            "Invalid configuration: 'queue' must be non-null.",
+            null,
+            (java.util.function.Consumer<Channel>)
+                channel -> {
+                  try {
+                    channel.basicConsume(null, null);
+                  } catch (Exception e) {
+                    throw new RuntimeException(e);
+                  }
+                }),
+        Arguments.of(
+            "<generated>",
+            "IOException",
+            null,
+            "receive",
+            (java.util.function.Consumer<Channel>)
+                channel -> {
+                  try {
+                    channel.basicGet("amq.gen-invalid-channel", true);
+                  } catch (Exception e) {
+                    throw new RuntimeException(e);
+                  }
+                }));
+  }
+
+  private static void rabbitSpan(
+      TraceAssert trace,
+      SpanDataAssert span,
+      int index,
+      String exchange,
+      String routingKey,
+      String operation,
+      String resource) {
+    rabbitSpan(trace, span, index, exchange, routingKey, operation, resource, null);
+  }
+
+  private static void rabbitSpan(
+      TraceAssert trace,
+      SpanDataAssert span,
+      int index,
+      String exchange,
+      String routingKey,
+      String operation,
+      String resource,
+      SpanData parentSpan) {
+    rabbitSpan(
+        trace,
+        span,
+        index,
+        exchange,
+        routingKey,
+        operation,
+        resource,
+        parentSpan,
+        null,
+        null,
+        null,
+        false,
+        false);
+  }
+
+  // Ignoring deprecation warning for use of SemanticAttributes
+  @SuppressWarnings("deprecation")
+  private static void rabbitSpan(
```

Comment:
The name of this is a little weird. Maybe rename to `verifySpan` or something. As it is now, `rabbitSpan` makes me think it might be returning a rabbit span.

I also think that breaking this method up by factoring out several helper methods could improve readability.

Reply:
I've refactored some of it out, hopefully making it easier to read